### PR TITLE
v4 feed update for .net 9

### LINKED ISSUE #590: Update tooling feed with released .NET 9 RTM templates
Update the tooling feed to reference .NET 9 RTM templates.

- [x] Update PR to reference new templates
- [x] Validate feed in PR branch
- [x] Merge changes

### PR DESCRIPTION
**Pull request by liliankasem**
Difference between 102 and 103: https://gist.github.com/liliankasem/695d6bc78cd0f6e24fffdf7fd4e751b8/revisions

GH Issue: https://github.com/Azure/azure-functions-tooling-feed/issues/590

## ⚠️ This PR is waiting on the new template releases first

### CODE REVIEW

File: cli-feed-v4.json [hunk 1]
```diff
@@ -34113,6 +34113,299 @@
...
+              "FUNCTIONS_EXTENSION_VERSION": "~4",
+              "FUNCTIONS_WORKER_RUNTIME": "dotnet-isolated"
+            },
+            "windowsSiteConfig": {
+              "netFrameworkVersion": "v6.0"
+            }
+          }
+        }
+      },
+      "coreTools": [ // Update all core tools values as well?
```
Comment (liliankasem) [inline cli-feed-v4.json]: ?

File: cli-feed-v4.json [hunk 2]
```diff
@@ -34113,6 +34113,299 @@
...
+              "hidden": false,
+              "displayVersion": "v4",
+              "targetFramework": ".NET 9",
+              "description": "Isolated LTS", // add LTS
+              "endOfLifeDate": "2026-05-12T00:00:00Z"
+            },
+            "capabilities": "isolated,net6,net7,net8,net9",
+            "sdk": {
+              "name": "Microsoft.Azure.Functions.Worker.Sdk",
+              "version": "2.0.0" // remove preview1
```
Comment (liliankasem) [inline cli-feed-v4.json]: ^

File: cli-feed-v4.json [hunk 3]
```diff
@@ -34113,6 +34113,299 @@
...
+              "linuxFxVersion": "DOTNET-ISOLATED|8.0"
+            }
+          },
+          "net9-isolated": {
+            "displayInfo": {
+              "displayName": ".NET 9.0",
+              "hidden": false,
+              "displayVersion": "v4",
+              "targetFramework": ".NET 9",
+              "description": "Isolated LTS", // add LTS
```
Comment (liliankasem) [inline cli-feed-v4.json]: ^

File: cli-feed-v4.json [hunk 4]
```diff
@@ -34113,6 +34113,299 @@
...
+              "targetFramework": ".NET 8",
+              "description": "Isolated", // remove LTS
+              "endOfLifeDate": "2026-11-10T00:00:00Z"
+            },
+            "capabilities": "isolated,net6,net7,net8",
+            "sdk": {
+              "name": "Microsoft.Azure.Functions.Worker.Sdk",
+              "version": "1.16.2" // Update to latest?? shouldn't this be 1.18.1? or is this the first version net 8 support was introduced
+            },
+            "default": false, // change to false
```
Comment (liliankasem) [inline cli-feed-v4.json]: ^

File: cli-feed-v4.json [hunk 5]
```diff
@@ -34113,6 +34113,299 @@
...
+              "hidden": false,
+              "displayVersion": "v4",
+              "targetFramework": ".NET 8",
+              "description": "Isolated", // remove LTS
+              "endOfLifeDate": "2026-11-10T00:00:00Z"
+            },
+            "capabilities": "isolated,net6,net7,net8",
+            "sdk": {
+              "name": "Microsoft.Azure.Functions.Worker.Sdk",
+              "version": "1.16.2" // Update to latest?? shouldn't this be 1.18.1? or is this the first version net 8 support was introduced
```
Comment (liliankasem) [inline cli-feed-v4.json]: ?

File: cli-feed-v4.json [hunk 6]
```diff
@@ -34113,6 +34113,299 @@
...
+              "linuxFxVersion": "DOTNET-ISOLATED|7.0"
+            }
+          },
+          "net8-isolated": {
+            "displayInfo": {
+              "displayName": ".NET 8.0",
+              "hidden": false,
+              "displayVersion": "v4",
+              "targetFramework": ".NET 8",
+              "description": "Isolated", // remove LTS
```
Comment (liliankasem) [inline cli-feed-v4.json]: ^

File: cli-feed-v4.json [hunk 7]
```diff
@@ -34113,6 +34113,299 @@
...
+              "targetFramework": ".NET 9",
+              "description": "Isolated LTS", // add LTS
+              "endOfLifeDate": "2026-05-12T00:00:00Z"
+            },
+            "capabilities": "isolated,net6,net7,net8,net9",
+            "sdk": {
+              "name": "Microsoft.Azure.Functions.Worker.Sdk",
+              "version": "2.0.0" // remove preview1
+            },
+            "default": true, // change to true
```
Comment (liliankasem) [inline cli-feed-v4.json]: ^

File: cli-feed-v4.json (hunk 1 above)
Comment (mattchenderson) [reply cli-feed-v4.json]: These do not need to be touched.

File: cli-feed-v4.json [hunk 8]
```diff
@@ -41,12 +41,12 @@
       "hidden": false
     },
     "v4": {
-      "release": "4.101.0",
+      "release": "4.101.0", // Update to 102 after validation of prerelease first
       "releaseQuality": "GA",
       "hidden": false
     },
     "v4-prerelease": {
```
Comment (mattchenderson) [inline cli-feed-v4.json]: these can move in lockstep. Validation can be done fully against the PR branch, and we don't need to go through the two-phase process

File: cli-feed-v4.json (hunk 6 above)
Comment (mattchenderson) [reply cli-feed-v4.json]: This should not change.

File: cli-feed-v4.json (hunk 5 above)
Comment (mattchenderson) [reply cli-feed-v4.json]: This should move to 2.0.0 to match .NET 9

File: cli-feed-v4.json (hunk 4 above)
Comment (mattchenderson) [reply cli-feed-v4.json]: This should not change.

File: cli-feed-v4.json (hunk 7 above)
Comment (mattchenderson) [reply cli-feed-v4.json]: This should not change.

File: cli-feed-v4.json (hunk 3 above)
Comment (mattchenderson) [reply cli-feed-v4.json]: This is not an LTS release. The correct change for this line is to simply remove the Preview designation.
```suggestion
              "description": "Isolated",
```

**Review by mattchenderson — CHANGES_REQUESTED** [review]
Unlinked the release item as merge of this PR is not sufficient to call that done. Please also provide a diff gist to make it easier for folks to review the changes between the two versions.

**Comment by liliankasem** [comment]
> Unlinked the release item as merge of this PR is not sufficient to call that done. Please also provide a diff gist to make it easier for folks to review the changes between the two versions.

Here is the diff between 4.102.0 and 4.103.0:

https://gist.github.com/liliankasem/695d6bc78cd0f6e24fffdf7fd4e751b8/revisions

File: cli-feed-v4.json [hunk 9]
```diff
@@ -34526,6 +34526,419 @@
...
+            },
+            "capabilities": "net6",
+            "sdk": {
+              "name": "Microsoft.NET.Sdk.Functions",
+              "version": "4.1.1"
+            },
+            "default": false,
+            "localEntryPoint": "func.exe",
+            "targetFramework": "net6.0",
+            "itemTemplates": "https://www.nuget.org/api/v2/package/Microsoft.Azure.WebJobs.ItemTemplates/4.0.5050",
```
Comment (liliankasem) [inline cli-feed-v4.json:34551]: I'm just guessing the new release version here, but will make sure to double check and update once these templates are published for real

File: cli-feed-v4.json [hunk 10]
```diff
@@ -34526,6 +34526,419 @@
...
+              "hidden": false,
+              "displayVersion": "v4",
+              "targetFramework": ".NET 8",
+              "description": "Isolated LTS",
+              "endOfLifeDate": "2026-11-10T00:00:00Z"
+            },
+            "capabilities": "isolated,net6,net7,net8",
+            "sdk": {
+              "name": "Microsoft.Azure.Functions.Worker.Sdk",
+              "version": "2.0.0"
```
Comment (surgupta-msft) [inline cli-feed-v4.json:34687]: Wanted to double check this. Do we want sdk 2.0.0 for .net8 also? @mattchenderson

Comment (surgupta-msft) [reply cli-feed-v4.json:34687]: Looks like "yes" as per this comment - https://github.com/Azure/azure-functions-tooling-feed/pull/592#discussion_r1831780266

Comment (liliankasem) [reply cli-feed-v4.json:34687]: > Looks like "yes" as per this comment - [#592 (comment)](https://github.com/Azure/azure-functions-tooling-feed/pull/592#discussion_r1831780266)

Yeah this was my understanding as well based on @mattchenderson's feedback

File: cli-feed-v4.json (hunk 9 above)
Comment (liliankasem) [reply cli-feed-v4.json:34551]: Updated to 5051

**Review by jviau — APPROVED** [review]

**Review by surgupta-msft — APPROVED** [review]

**Review by mattchenderson — APPROVED** [review]

**Review by fabiocav — APPROVED** [review]

**Pull request merged**